### DISCUSSION TITLE
Skip dependency updates affecting only specific projects

### DISCUSSION THREAD

**Discussion by aKzenT** [renovatebot/renovate#16683] [Request Help]
Is it possible to skip dependency updates if they only affect specific projects?

E.g. imagine I have some test project whose dependencies I don't care about. So I don't want to update them automatically or update them only in larger intervals. However if I have a dependency in my main project that also happens to be a dependency in the test project, then I want to keep the dependencies in sync. So the PR should include the updates for both projects.

So basically:
If dependency update would only affect project A => Skip Update
If dependency update affects project A & B => Update A & B

What I tried was using matchPaths to exclude changes for project A and then include changes again for project B.
```
    {
      "matchPaths": [ "A/" ],
      "enabled": false
    },
    {

      "matchPaths": [ "B/" ],
      "enabled": true,
    },
```

It seems however that the rules are evaluated per individual dependency update, so that the effect is that the updated to project A are simply excluded from the PR.

So is there currently a way to do this? Or would this be a feature request?

**Comment by hasanwhitesource** [comment]
To make sure of what is the set up here.
You have two separate repos A and B. If there are some X dependencies in A that renovate updated you also want those X dependencies to be updated in B? 
Or do you have a mono repo having A and B under it ?

**Reply by aKzenT** [reply]
It's a monorepo. The actual use case is a bit more complicated involving two teams working on two projects and a shared project between two teams, but hopefully the example is close enough.

So if there are updates in B, I also want to update them in A, but not the other way arround.

**Reply by PhilipAbed** [reply]
you can either Enable or Disable a match of packages/files/paths,
 but i dont think you can Disable a Repository/file/path and then enable it Conditionally

**Comment by PhilipAbed** [comment]
in my opinion a suggested scripted solution for your case would be
Make 2 runs.
first make a dry run then the second run should run conditionally, The idea is to Understand if you have updates for A first.
then after that you can decide if you wanna run at all .

Run renovate in [dryRun](https://docs.renovatebot.com/self-hosted-configuration/#dryrun) flag, `dryRun: lookup`

then parse the logs and see what the dryRun prints for updates, if you see any updates for repository A
then you run again without the `dryRun`

its a workaround solution, but it should work.

hope that helps

**Reply by PhilipAbed** [reply]
this will not work for you 100% i guess though, you cant Choose which dependencies to update, unless you add them to a packageRule which makes things more complicated for you...
`if Dependency Doesnt exist in B but exists in  A then ignore it.. put it in the Disabled rule`
renovate doesnt support your case directly i guess

**Reply by aKzenT** [reply]
Hm. That's unfortunate. Thank you for your insights!
